### PR TITLE
Task 42671 : Change the UI of the filter field

### PR DESCRIPTION
**Pull request by Rdidmohamedelhadi**

### CODE REVIEW

**Review by boubaker — APPROVED** [review]

File: webapp/portlet/src/main/webapp/common/components/ExoIdentitySuggester.vue [hunk 1]
```diff
@@ -182,6 +183,12 @@ export default {
         return null;
       },
     },
+    innerIcon: {
```
Comment (boubaker) [inline webapp/portlet/src/main/webapp/common/components/ExoIdentitySuggester.vue]: We will have to differentiate the display of the suggester input by using an explicit and meaningful variable name like `filterStyle` prop of type boolean that will have `false` as default value.

File: webapp/portlet/src/main/webapp/common/components/ExoIdentitySuggester.vue [hunk 2]
```diff
@@ -15,7 +15,8 @@
       :height="height"
       :filter="filterIgnoredItems"
       :hide-no-data="hideNoData"
-      :class="(required && !value && 'required-field invalid') || (required && 'required-field')"
+      :class="(required && !value && 'required-field invalid') || (required && 'required-field') || (innerIcon && 'customIdentitySuggester' && 'customIconIdentitySuggester') || (!innerIcon && 'identitySuggesterStyle')"
```
Comment (boubaker) [inline webapp/portlet/src/main/webapp/common/components/ExoIdentitySuggester.vue]: Add a computed property to : 
1/ use required style or not
2/ use `identitySuggesterFilterStyle` class

```vue.js
computed: {
  autocompleteClass() {
    const requiredClass = (this.required && !this.value && 'required-field invalid') || (this.required && 'required-field') || '';
    const sugesterStyleClass = this.filterStyle && 'identitySuggesterFilterStyle' || 'identitySuggesterInputStyle';
    return `${requiredClass} ${sugesterStyleClass}`;
  }
}
```

File: webapp/portlet/src/main/webapp/common/components/ExoIdentitySuggester.vue [hunk 3]
```diff
@@ -15,7 +15,8 @@
...
+      :prepend-inner-icon="innerIcon ? 'fa-filter' : ''"
```
Comment (boubaker) [inline webapp/portlet/src/main/webapp/common/components/ExoIdentitySuggester.vue]: Same here, you can use a computed property

File: webapp/portlet/src/main/webapp/common/components/ExoIdentitySuggester.vue [hunk 4]
```diff
@@ -195,6 +202,14 @@ export default {
     };
   },
   computed: {
+    icon() {
```
Comment (boubaker) [inline webapp/portlet/src/main/webapp/common/components/ExoIdentitySuggester.vue]:
```suggestion
    prependInnerIcon() {
```

File: webapp/portlet/src/main/webapp/common/components/ExoIdentitySuggester.vue [hunk 5]
```diff
@@ -182,6 +183,12 @@ export default {
         return null;
       },
     },
+    filterStyle: {
+      type: Boolean,
+      default: function () {
+        return false;
+      },
```
Comment (boubaker) [inline webapp/portlet/src/main/webapp/common/components/ExoIdentitySuggester.vue]: When it's boolean, you can avoid declaring the default value as a function
```suggestion
      default: false,
```

Comment (Rdidmohamedelhadi) [reply webapp/portlet/src/main/webapp/common/components/ExoIdentitySuggester.vue]: done

**Review by boubaker — APPROVED** [review]

**Pull request merged**